src/pages/Facility/settings/devices/components/DeviceServiceHistory.tsx

```diff
@@ -0,0 +1,156 @@
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { format } from "date-fns";
+import { Edit, Plus } from "lucide-react";
+import { useQueryParams } from "raviger";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import { cn } from "@/lib/utils";
+
+import { Button } from "@/components/ui/button";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+
+import PaginationComponent from "@/components/Common/Pagination";
+import { TableSkeleton } from "@/components/Common/SkeletonLoading";
+
+import { RESULTS_PER_PAGE_LIMIT } from "@/common/constants";
+
+import query from "@/Utils/request/query";
+import { ServiceHistory } from "@/types/device/device";
+import deviceApi from "@/types/device/deviceApi";
+
+import ServiceHistorySheet from "./ServiceHistorySheet";
+
+interface DeviceServiceHistoryProps {
+  facilityId: string;
+  deviceId: string;
+}
+
+export default function DeviceServiceHistory({
+  facilityId,
+  deviceId,
+}: DeviceServiceHistoryProps) {
+  const { t } = useTranslation();
+  const [isServiceSheetOpen, setIsServiceSheetOpen] = useState(false);
```

Comment:
Let's maintain different sheets (add service, edit service sheet) and use sheet trigger instead.

Refer to AddUserSheet and EditUserSheet for examples.